text/0004-optional-frontend.md

```diff
@@ -0,0 +1,45 @@
+- Title: Optional Meilisearch Front-end
+- Start Date: 2020/11/16
+- specification PR:
+- Meilisearch Issue: #411
+
+# Optional Meilisearch Front-end
+
+## First section: Feature Description and Interaction
+
+### Summary
+
+For testing purposes, MeiliSearch is shipped with a frontend that is served on the server's root. This provides the user with a basic GUI to perform searches on his document collection. While this is useful in a development environment, this is not necessarily desirable in a production environment. This specification propose to make this frontend optional thanks to binary setting.
+
+### Motivation
+
+In most production cases, there is no need for this frontend, and exposing it is a bad practice. We want to provide an opt-out option for this feature.
+
+### Prior Art and R&D
+
+N.A
+
+### Explanation
+
+On MeiliSearch startup, we would like to pass a flag `--disable-web-gui` that disables the web user interface.
```

Comment:
What do we do then? @curquiza @LegendreM

Reply:
I incline to the @LegendreM suggestion that automatically disables it when the production mode is set.  Since exposing the front-end is a bad practice in production, disable it automatically could help our users, otherwise all the MeiliSearch production should have both settings: `--env production` and `--disable-web-gui`. `--env production` should be enough.
Plus, ut would enforce users to use the production mode and so that we could know who is really using MeiliSearch in production 🙂